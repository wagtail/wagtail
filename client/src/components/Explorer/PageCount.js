import PropTypes from 'prop-types';
import React from 'react';

import { ADMIN_URLS, STRINGS } from '../../config/wagtailConfig';
import Icon from '../Icon/Icon';

const PageCount = ({ page }) => {
  const count = page.children.count;

  return (
    <a
      href={`${ADMIN_URLS.PAGES}${page.id}/`}
      className="c-explorer__see-more"
      tabIndex={0}
    >
      {STRINGS.SEE_ALL}
      <span>{` ${count} ${count === 1 ? STRINGS.PAGE.toLowerCase() : STRINGS.PAGES.toLowerCase()}`}</span>
      <Icon name="arrow-right" />
    </a>
  );
};

PageCount.propTypes = {
  page: PropTypes.object.isRequired,
};

export default PageCount;
