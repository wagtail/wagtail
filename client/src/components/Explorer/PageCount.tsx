import React from 'react';

import { ADMIN_URLS } from '../../config/wagtailConfig';
import Icon from '../Icon/Icon';

interface PageCountProps {
  page: {
    id: number;
    children: {
      count: number;
    };
  };
}

const PageCount: React.FunctionComponent<PageCountProps> = ({ page }) => {
  const count = page.children.count;

  return (
    <a href={`${ADMIN_URLS.PAGES}${page.id}/`} className="c-explorer__see-more">
      {gettext('See all')}
      <span>{` ${count} ${
        count === 1
          ? gettext('Page').toLowerCase()
          : gettext('Pages').toLowerCase()
      }`}</span>
      <Icon name="arrow-right" />
    </a>
  );
};

export default PageCount;
