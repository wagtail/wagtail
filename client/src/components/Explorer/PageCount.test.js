import React from 'react';
import { shallow } from 'enzyme';

import PageCount from './PageCount';

const mockProps = {
  page: {
    id: 1,
    children: {
      count: 1,
    },
  },
};

describe('PageCount', () => {
  it('exists', () => {
    expect(PageCount).toBeDefined();
  });

  it('works', () => {
    expect(shallow(<PageCount {...mockProps} />)).toMatchSnapshot();
  });

  it('plural', () => {
    const props = Object.assign({}, mockProps);
    props.page.children.count = 5;
    expect(shallow(<PageCount {...props} />)).toMatchSnapshot();
  });
});
