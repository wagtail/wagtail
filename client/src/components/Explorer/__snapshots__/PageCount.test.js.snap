// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageCount plural 1`] = `
<a
  className="c-explorer__see-more"
  href="/admin/pages/1/"
  tabIndex={0}
>
  See all
  <span>
     5 pages
  </span>
  <Icon
    className={null}
    name="arrow-right"
    title={null}
  />
</a>
`;

exports[`PageCount works 1`] = `
<a
  className="c-explorer__see-more"
  href="/admin/pages/1/"
  tabIndex={0}
>
  See all
  <span>
     1 page
  </span>
  <Icon
    className={null}
    name="arrow-right"
    title={null}
  />
</a>
`;
